samplesheets/views_api.py

```diff
@@ -310,6 +316,182 @@ def post(self, request, *args, **kwargs):
         return Response(ret_data, status=status.HTTP_200_OK)
 
 
+class IrodsAccessTicketListAPIView(SODARAPIBaseProjectMixin, ListAPIView):
+    """
+    List iRODS access tickets for a project.
+
+    **URL:** ``/samplesheets/api/irods/ticket/list/{Project.sodar_uuid}``
+
+    **Methods:** ``GET``
+
+    **Query parameters:**
+
+    - ``active`` (boolean, default ``0``)
+    """
+
+    permission_required = 'samplesheets.edit_sheet'
+    serializer_class = IrodsAccessTicketSerializer
+
+    def get_queryset(self):
+        project = self.get_project()
+        tickets = IrodsAccessTicket.objects.filter(
+            study__investigation__project=project
+        )
+        active = self.request.query_params.get('active', '0')
+        active = bool(int(active))
+        if active:
+            tickets = [t for t in tickets if t.is_active()]
+        return tickets
+
+
+class IrodsAccessTicketRetrieveAPIView(
+    SODARAPIGenericProjectMixin, RetrieveAPIView
+):
+    """
+    Retrieve an iRODS access ticket for a project.
+
+    **URL:** ``/samplesheets/api/irods/ticket/retrieve/{IrodsAccessTicket.sodar_uuid}``
+
+    **Methods:** ``GET``
+    """
+
+    lookup_field = 'sodar_uuid'
+    lookup_url_kwarg = 'irodsaccessticket'
+    permission_required = 'samplesheets.edit_sheet'
+    serializer_class = IrodsAccessTicketSerializer
+
+    def get_object(self):
```

Comment:
The current dev version of SODAR is based on the latest dev commit of SODAR Core, where I have introduced a way to get over requiring this override.

It should work if you remove the `get_object()` override and add the following to your class:

`queryset_project_field = 'study__investigation__project'`